app/controllers/tweets_controller.rb

```diff
@@ -53,6 +50,8 @@ def setup_tweet
   def show
     @replies = Tweet.get_replies(params[:id])
     @user = User.find(@tweet.user_id)
+    Visitor.create(tweet_id: @tweet.id, user_id: current_user.id) if Visitor.where(tweet_id: @tweet.id,
+                                                                                   user_id: current_user.id).blank?
```

Comment:
- This code can be moved to model.
- This will even create Visitor record for tweet-author
- This can be written more simple way. refer https://guides.rubyonrails.org/active_record_querying.html#find-or-build-a-new-object